pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -381,13 +406,16 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 
 	}
 
-	if _, ok := data.(*getRepoDataBySearchResponse); ok {
+	if searchRepos != nil {
 
 		var wg1 sync.WaitGroup
 		var opBuf int = 3
 
 		chunkSize := (len(searchRepos) + opBuf - 1) / opBuf
```

Comment:
This part is just checking if the previous section that gets us all of the repos we want to process data from actually returned anything and if so, continue but otherwise dont bother with the go routine stuff